typescript/core/src/McapWriter.ts

```diff
@@ -130,7 +192,9 @@ export class McapWriter {
   }
 
   async end(): Promise<void> {
-    await this.#finalizeChunk();
+    if (!this.#appendMode) {
+      await this.#finalizeChunk();
```

Comment:
Why don't we want to do this in append mode? It seems like this would lead to messages not being written out.